File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        max_count = 0
        flipcount = 0
        l=0
        for r in range(len(nums)):
            if (nums[r]!=1):
                flipcount+=1
            while flipcount>k:
                if nums[l] == 0:
                    flipcount-=1
                l+=1
            max_count=max(max_count,r-l+1)

        return max_count

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
       int count =0,ans=0,l=0;
       for(int r =0;r<nums.length;r++) {
            if(nums[r]!=1){
                count+=1;
            }
            while(count>k){
                if(nums[l]==0) count --;
                l++;
            }
            ans = Math.max(ans,r-l+1);
       }
       return ans;
    }
}